src-electron/generator/helper-zcl.js

```diff
@@ -2233,27 +2233,46 @@ async function if_mfg_specific_cluster(clusterId, options) {
  */
 async function as_generated_default_macro(value, attributeSize, options) {
   let default_macro_signature = ''
+  let temp = ''
   if (attributeSize > 2) {
-    let default_macro = helperC
-      .asHex(value, null, null)
-      .replace('0x', '')
-      .match(/.{1,2}/g)
+    // String value
+    if (isNaN(value)) {
+      return format_zcl_string_as_characters_for_generated_defaults(
```

Comment:
This is not used for Matter, I assume?  If it is, it's not handling long strings correctly.....

Reply:
Yes this is not used by matter. Why do you say that?

Reply:
`format_zcl_string_as_characters_for_generated_defaults` assumes the string length is a singly byte.  For long strings it's two bytes.